pkg/identityserver/user_registry.go

```diff
@@ -764,6 +770,12 @@ func (is *IdentityServer) purgeUser(ctx context.Context, ids *ttnpb.UserIdentifi
 		if err != nil {
 			return err
 		}
+		if err := st.PurgeEntityBookmarks(ctx, ids.GetEntityIdentifiers()); err != nil {
```

Comment:
Is this correct ? You seem to be purging the entity bookmarks _for all users_ when _a single user_ is deleted.

Reply:
This is probably a naming issue, I got used to writing it like this but it does give the wrong impression of what is happening.

`PurgeEntityBookmarks` in this case is not purging the entity bookmarks for all users. It purges bookmarks to which `(<entity_type>, <entity_id>)` are present on the `user_bookmark` table.

Will change the names to be something like `PurgeBookmarksByEntityID` and the user operations to `PurgeBookmarksByUserID`.

Reply:
I don't follow. We are in `purgeUser` now. Take the following situation:

1. User 1 bookmarks entity 1.
2. User 2 bookmarks entity 1.
3. We delete and purge user 2.
4. At this point you also have purged the bookmark for user 1. Why ?

Reply:
I still believe its just a misunderstanding regarding the naming of the method, or maybe a misunderstanding to the possibility for an user to bookmark another user.

>User 1 bookmarks entity 1.
>User 2 bookmarks entity 1.
>We delete and purge user 2.
>At this point you also have purged the bookmark for user 1. Why ?

When we purge the user2, `PurgeEntityBookmarks` deletes rows where:
```
return q.Where(
    "?TableAlias.entity_type = ? AND ?TableAlias.entity_id = ?",
    ids[0].EntityType(),
    ids[0].IDString(),
)
```

So the bookmark for the user-1 and the entity-1 still exists. I even tested locally just to be completely sure and not be something from my head.

The meaning behind `PurgeEntityBookmarks` is that it deletes all bookmarks to which the entity is being referenced to, including an user. So `PurgeEntityBookmarks` call in the `purgeUser` serves purely to delete references of other users to the user being purged.

Reply:
I understand now what happens, you are correct. It also makes sense that it is like this, I just didn't realize that users can be bookmarked too, and in this case `ids` is the identifiers of a user.

The naming can stay as is, I think it is fine in retrospective.